test/TrueLayer.Tests/TrueLayerServiceCollectionExtensionsTests.cs

```diff
@@ -35,5 +35,38 @@ public void Should_register_truelayer_client()
             var client = services.GetRequiredService<ITrueLayerClient>();
             client.Should().NotBeNull();
         }
+
+        [Fact]
+        public void Should_register_keyed_truelayer_clients()
+        {
+            const string gbpServiceKey = "GbpClient";
+            const string eurServiceKey = "EurClient";
+            const string privateKey = @"-----BEGIN EC PRIVATE KEY-----
+MIHcAgEBBEIALJ2sKM+8mVDfTIlk50rqB5lkxaLBt+OECvhXq3nEaB+V0nqljZ9c
+5aHRN3qqxMzNLvxFQ+4twifa4ezkMK2/j5WgBwYFK4EEACOhgYkDgYYABADmhZbj
+i8bgJRfMTdtzy+5VbS5ScMaKC1LQfhII+PTzGzOr+Ts7Qv8My5cmYU5qarGK3tWF
+c3VMlcFZw7Y0iLjxAQFPvHqJ9vn3xWp+d3JREU1vQJ9daXswwbcoer88o1oVFmFf
+WS1/11+TH1x/lgKckAws6sAzJLPtCUZLV4IZTb6ENg==
+-----END EC PRIVATE KEY-----";
+            var configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(new[]
+                {
+                    new KeyValuePair<string, string?>("TrueLayer:ClientId", "client_id"),
+                    new KeyValuePair<string, string?>("TrueLayer:ClientSecret", "secret"),
+                    new KeyValuePair<string, string?>("TrueLayer:Payments:SigningKey:KeyId", Guid.NewGuid().ToString()),
+                    new KeyValuePair<string, string?>("TrueLayer:Payments:SigningKey:PrivateKey", privateKey)
+                })
+                .Build();
+            var services = new ServiceCollection()
+                .AddKeyedTrueLayer(configuration, serviceKey: gbpServiceKey)
+                .AddKeyedTrueLayer(configuration, serviceKey: eurServiceKey)
+                .BuildServiceProvider();
+
+            var gbpClient = services.GetRequiredKeyedService<ITrueLayerClient>(gbpServiceKey);
+            var eurClient = services.GetRequiredKeyedService<ITrueLayerClient>(eurServiceKey);
+
+            gbpClient.Should().NotBeNull();
+            eurClient.Should().NotBeNull();
```

Comment:
Do we have to make sure these are made of the config that we expect? or overhead?

Reply:
I've made a minor improvement but unfortunately I don't think there's a easy way to do that.
we might consider these tests as a sort of smoke test for the registration